src/timesync/timesyncd-manager.c

```diff
@@ -787,7 +786,8 @@ int manager_connect(Manager *m) {
         if (!ratelimit_below(&m->ratelimit)) {
                 log_debug("Delaying attempts to contact servers.");
 
-                r = sd_event_add_time_relative(m->event, &m->event_retry, clock_boottime_or_monotonic(), RETRY_USEC, 0, manager_retry_connect, m);
+                r = sd_event_add_time_relative(m->event, &m->event_retry, clock_boottime_or_monotonic(), m->network_online_retry_timer,
```

Comment:
In my testing and reproing of the issue, this code path is not hit at all. The default 30s would not be a problem here - the problem is that we are hitting PollIntervalMaxSec - which is half an hour - and never backing down from there, because networkd is not used and thus we can't detect that the network is gone/back and react.

Reply:
In other words, in steady state we get poll_interval_usec all the way up to PollIntervalMaxSec, then the network goes away, and we have nothing to wake us up before it expires.

Reply:
Basing this on observing what happens and debug level logs:

```
Dec 08 14:22:32 debian systemd-timesyncd[3679]: Selected server 0.debian.pool.ntp.org.
Dec 08 14:22:32 debian systemd-timesyncd[3679]: Resolving 0.debian.pool.ntp.org...
Dec 08 14:22:38 debian systemd-timesyncd[3679]: Failed to resolve 0.debian.pool.ntp.org: Name or service not known
Dec 08 14:22:38 debian systemd-timesyncd[3679]: Selected server 1.debian.pool.ntp.org.
Dec 08 14:22:38 debian systemd-timesyncd[3679]: Resolving 1.debian.pool.ntp.org...
Dec 08 14:22:44 debian systemd-timesyncd[3679]: Failed to resolve 1.debian.pool.ntp.org: Name or service not known
Dec 08 14:22:44 debian systemd-timesyncd[3679]: Selected server 2.debian.pool.ntp.org.
Dec 08 14:22:44 debian systemd-timesyncd[3679]: Resolving 2.debian.pool.ntp.org...
Dec 08 14:22:50 debian systemd-timesyncd[3679]: Failed to resolve 2.debian.pool.ntp.org: Name or service not known
Dec 08 14:22:50 debian systemd-timesyncd[3679]: Selected server 3.debian.pool.ntp.org.
Dec 08 14:22:50 debian systemd-timesyncd[3679]: Resolving 3.debian.pool.ntp.org...
Dec 08 14:22:56 debian systemd-timesyncd[3679]: Failed to resolve 3.debian.pool.ntp.org: Name or service not known
Dec 08 14:22:56 debian systemd-timesyncd[3679]: Waiting after exhausting servers.
```

Reply:
Your testing is wrong. Netword is enabled by default

Reply:
It is not on Debian, which is where https://github.com/systemd/systemd/issues/16529 was reported. On Debian, ifupdown is used on headless installations and network-manager on desktops.

Reply:
You should test on system where netword is enabled else its pointless